target/linux/x86/image/grub-efi.cfg

```diff
@@ -6,7 +6,14 @@ set timeout="@TIMEOUT@"
 search -l kernel -s root
 
 menuentry "@TITLE@" {
-	linux /boot/vmlinuz @GPT_ROOTPART@ @CMDLINE@ noinitrd
+	linux /boot/vmlinuz @GPT_ROOTPART@ @CMDLINE@
+	if [ -s /boot/intel-ucode.img ]; then
+		initrd /boot/intel-ucode.img
+	fi
+	if [ -s /boot/amd-ucode.img ];then
+		initrd /boot/amd-ucode.img
+	fi
```

Comment:
Try this:
```
	if [ -s /boot/intel-ucode.img ]; then
		linux /boot/vmlinuz @GPT_ROOTPART@ @CMDLINE@
	elif [ -s /boot/amd-ucode.img ]; then
		linux /boot/vmlinuz @GPT_ROOTPART@ @CMDLINE@
	else
		linux /boot/vmlinuz @GPT_ROOTPART@ @CMDLINE@ noinitrd
	fi
	if [ -s /boot/intel-ucode.img ]; then
		initrd /boot/intel-ucode.img
	fi
	if [ -s /boot/amd-ucode.img ]; then
		initrd /boot/amd-ucode.img
	fi
```
Change the remaining cfg files similarly.

Reply:
Good suggestion, works for me on the Intel system.